src/util/matchToKartoffel.js

```diff
@@ -641,13 +641,13 @@ module.exports = async (origin_obj, dataSource, Auth, defaultSendLog, flowType)
             sendLog(logLevel.error, logDetails.error.ERR_UNIDENTIFIED_DATA_SOURCE);
     }
 
-
+    let newGroups;
     if (obj.hierarchy) {
         
-        obj.directGroup = await directGroupHandler(obj, Auth);
+        [obj.directGroup, newGroups] = await directGroupHandler(obj, Auth);
         
         delete obj.hierarchy;
     }
 
-    return obj;
+    return [obj, newGroups];
```

Comment:
consider the use of object destruction instead of an array, if possible it's preferable